stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java

```diff
@@ -0,0 +1,186 @@
+                    name = name.substring(index);
+                }
+                    if (entry.isDirectory()) {
+                        mkdirs(outdir, name);
+                        continue;
+                    }
+                    // this part is necessary because file entry can come before directory entry where the
+                    // file is located
+                    dir = getDirectoryPath(name);
+                    if (dir != null) {
+                        mkdirs(outdir, dir);
+                    }
+                    extractFile(zin, outdir, name);
+            }
+        } catch (IOException e) {
+            log.debug("I/O Exception when processing files ", e);
+            log.error("I/O Exception when processing files " + e.getMessage());
+        } finally {
+            try {
+                if (zin != null) {
+                    zin.close();
+                }
+            } catch (IOException e) {
+                log.debug("I/O Exception when closing the input stream ", e);
+                log.error("I/O Exception when closing the input stream " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Extract files from the zipInputStream.
+     *
+     * @param in     zipInputStream object
+     * @param outdir output directory file
+     * @param name   name of the file
+     */
+    private static void extractFile(ZipInputStream in, Path outdir, String name) {
+        byte[] buffer = new byte[4096];
+        BufferedOutputStream out = null;
+        try {
+            Path resourcePath = Paths.get(outdir.toString()).resolve(name);
+            out = new BufferedOutputStream(new FileOutputStream(resourcePath.toString()));
+            int count;
+            while ((count = in.read(buffer)) != -1) {
+                out.write(buffer, 0, count);
+            }
+        } catch (FileNotFoundException e) {
+            log.debug("File not found to process " + outdir, e);
+            log.error("File not found to process " + e.getMessage());
+        } catch (IOException e) {
+            log.debug("I/O Exception when closing the input stream " + outdir, e);
+            log.error("I/O Exception when closing the input stream " + e.getMessage());
+        } finally {
+            if (out != null) {
+                try {
+                    out.close();
+                } catch (IOException e) {
+                    log.debug("I/O Exception when closing the input stream ", e);
+                    log.error("I/O Exception when closing the input stream " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Create the directory name.
+     *
+     * @param name name of the directory
+     * @return directory name
+     */
+    private static String getDirectoryPath(String name) {
+        if (name != null) {
+            int s = name.lastIndexOf(File.separatorChar);
+            return s == -1 ? null : name.substring(0, s);
```

Comment:
I noticed this function do multiple returns of null and handled at higher layer? Can't we get rid of null return cases?